src/main/java/com/yanolja_final/domain/order/entity/Order.java

```diff
@@ -65,4 +72,29 @@ public Order(User user, Package aPackage, Review review, Long availableDateId,
         this.code = code;
         this.detailInfo = detailInfo;
     }
+
+    public static Order userOrderWithEarliestDepartureDate(User user) {
+        List<Order> userOrders = user.getOrders();
+        Order userOrderWithEarliestDepartureDate = userOrders.stream()
+            .min(Comparator.comparing(order -> order.departureDateForOrder(order.getPackageDepartureOption())))
+            .orElseThrow(OrderNotFoundException::new);
+        return userOrderWithEarliestDepartureDate;
+    }
+
+    private LocalDate departureDateForOrder(PackageDepartureOption option) {
+        return option.getDepartureDate();
+    }
+
+    public PackageDepartureOption getPackageDepartureOption() {
+        List<PackageDepartureOption> availableDates = aPackage.getAvailableDates();
+        if (availableDates.isEmpty()) {
+            throw new PackageDateNotFoundException();
+        }
+        Long availableDateId = this.availableDateId;
+        PackageDepartureOption packageDepartureOption = availableDates.stream()
+            .filter(option -> option.getId().equals(availableDateId))
+            .findFirst()
+            .orElseThrow(PackageNotFoundException::new);
+        return packageDepartureOption;
```

Comment:
```suggestion
        return aPackage.getAvailableDates().stream()
            .filter(option -> option.getId().equals(this.availableDateId))
            .findFirst()
            .orElseThrow(PackageNotFoundException::new);
```

이렇게해도 지금 짜신 코드와 완전히 동일하게 작동합니다

1. availableDates가 Empty인 경우는 orElseThrow()에서 이미 잡습니다
2. availableDateId는 굳이 지역변수로 또 선언할 필요 없어보입니다.

Reply:
Package에 해당하는 PackageDepartureOption이 없는 케이스에 PackageNotFoundException조금 어색한거 같습니다!
-> PackageDepartureNotFoundException이 존재하니 사용하시면 될거 같습니다.

Reply:
![FeedBack1_PreventPassedDepartureException](https://github.com/yanolja-finalproject/Backend/assets/129931655/5c7c5edb-9cc9-43b7-b64b-a31f2a3c341a)

커밋 메시지는 남겨 두었지만, 기존 PackageDepartureOption을 불러올 용도로 만들어두었던 메서드에
지나간 날짜의 예외를 처리하는 로직을 추가 하였습니다.
// 추가로 기존 getDepartureDateForOrder 라는 기존에 Order를 위했던 메서드는 불필요한 것 같아서 지웠습니다.